sql/core/src/test/scala/org/apache/spark/sql/streaming/MapGroupsWithStateSuite.scala

```diff
@@ -83,6 +88,34 @@ class MapGroupsWithStateSuite extends StateStoreMetricsTest with BeforeAndAfterA
     }
   }
 
+  test("KeyedState - isTimingOut, setTimeoutDuration") {
+    import KeyedStateImpl._
+    var state: KeyedStateImpl[String] = null
+    state = new KeyedStateImpl[String](None)
+
+    assert(state.isTimingOut === false)
+    assert(state.getTimeoutTimestamp === TIMEOUT_TIMESTAMP_NOT_SET)
+    intercept[UnsupportedOperationException] {
+      state.setTimeoutDuration(1000)
+    }
+    intercept[UnsupportedOperationException] {
+      state.setTimeoutDuration("1 day")
+    }
+    assert(state.getTimeoutTimestamp === TIMEOUT_TIMESTAMP_NOT_SET)
+
+    state = new KeyedStateImpl[String](None, 1000, isTimeoutEnabled = true, isTimingOut = false)
```

Comment:
Judging by the tests below, it seems that timeout timestamps are `timeout + batchInterval`, which makes total sense but it might be worth adding to documentation that time-based triggers are affected by "system-latency" or something.. I believe this is true for most streaming frameworks anyway, but this "disclaimer" might avoid user confusion as he might expect his output after 200msec while internal batch interval is 250msec.
I'm only noting this because It caught me off-guard, until i saw the 1000msec batchInterval passed to `KeyedStateImpl` constructor.